proxy/coco/src/peer.rs

```diff
@@ -99,6 +102,16 @@ impl Api {
         api.paths().git_dir().join("")
     }
 
+    /// Get the [Seed] representation for this [Api] instance.
+    #[must_use]
+    pub fn seed(&self) -> Seed {
```

Comment:
An impl for `From` would be the more elegant solution for this.

Reply:
The `From` was a bit of a pain because `PeerApi` is parameterised over an `S` which needs to be a `Signer` et al. But I think I still agree, `From` would be good :+1: Just follow the trait bounds in `librad` on `PeerApi` to get the incantations right.

Reply:
`From` would be more elegant, but as pointed out it's not trivial and offers no further benefit compared to what's already in. Could leave this for last but don't see the point all things considered.

Reply:
Would something like this not be sufficient:

``` rust

impl From<&Api> for Seed {
    fn from(api: &Api) -> Self {
        Self {
            addr: api.listen_addr(),
            peer_id: api.peer_id(),
        }
    }
}
```

Reply:
You are right, @xla! When Fintan and I were pairing on this we were operating on the underlying `PeerApi`, which is parameterised, but we didn't think of implementing it for `Api` instead. Done!

Reply:
Well, I did. But I just didn't want to interact with the `lock` in a `From` impl :sweat_smile: